### DISCUSSION TITLE
Using separate bottles for Steam games

### DISCUSSION THREAD

**Discussion by gpanders** [Whisky-App/Whisky#652] [Q&A] [answered]
The [wiki](https://github.com/Whisky-App/Whisky/wiki/A-Hitchhiker's-Guide-to-Whisky#when-should-i-make-a-new-bottle) recommends using separate bottles for each game, but I am curious how this is meant to be done for games downloaded through Steam. Is the recommendation here to download and configure Steam separately in each bottle? Or should there be one Steam bottle, and then downloaded games can be copied into other bottles individually?

The [Game Support](https://github.com/Whisky-App/Whisky/wiki/Game-Support) wiki page also has the instructions:

>* Create a new Windows 10 bottle
>* Install in Steam as normal

But it's still not clear to me here what "as normal" means. Does this mean install Steam in the new bottle and download the game there?

**Comment by muradbu** [comment]
I'm wondering the same. I'm new to Wine and never used CrossOver so I'm not sure what to do here.

**Comment by Morgan-6Freedom** [comment]
same question

**Comment by billiamg** [comment]
I think what is meant by this is (at least what I've done to get this working) is:

1. Make a bottle for the game you're going to play in steam
2. Add a new drive to this bottle for your steam library. In Whisky go to 'Bottle Configuration' > 'Open Wine Configuration' > 'Drives' > 'Add..' a new drive and give it a letter (I picked S:) > Set 'Path:' to a folder in your user folder, for example /User/username/SteamLibrary
3. Install steam in this bottle
4. Launch steam, go to 'Steam' > 'Settings' > 'Storage' > 'Add Drive' at the top (click on the existing 'Local Drive (C:)') > Select the drive letter you picked (S: in my case)
5. Install the game that this bottle is for to this steam library (S: in my case)

Repeat these steps for each new game.

There might be a better way... happy to be corrected with the intended method.

**Reply by muradbu** [reply]
But why do you add a new drive for each bottle? I just create a new bottle for each game and install Steam and said game in said bottle. Or am I misunderstanding, and you're using a single drive for games across multiple bottles?

**Reply by billiamg** [reply]
My approach means that there's a single game library and was my interpretation of the text on the [wiki page](https://docs.getwhisky.app)

> For Steam games, you can create one shared Windows game library that you use between all of your bottles. This helps keep your game installations detached from your bottle instances, so you won't have to re-download everything if your bottle mucks up.

Your approach also addresses this, however if a bottle 'mucks up' you'll need to grab the files in finder and move them to the new bottle. I think either approach is valid, would love for the wiki writer to clarify what they mean though.

**Reply by muradbu** [reply]
Oh I see. That part wasn't in the docs before they made a dedicated website I think, so I must've missed it. Anyhow, it does sound like a nicer solution to have 1 drive for all games.

**Reply by malice00** [reply]
Just found this discussion. Any word from the dev/writer of the documentation on this? I mean, your suggestion might work (haven't tried yet), but it would mean I can only have around 20 games in my library!?

I say 20, because in the configuration I see that I already have 4 drives connected (no configuration from my side here!) and afaik Windows doesn't like having A and B not being a disk drive (it used to be that way, but I haven't used Windows in ages, so this may have changed).

Anyway, personally I would've liked some way to mount bottles as directories or something similar, so I can actually have some more games in the library -- I'm pretty sure most of us have more than 20 games on Steam...
Of course, I could always switch drives, but then I would have to remember where they were mounted and I would probably have to restart Steam constantly.

**Reply by muradbu** [reply]
AFAIK no official word from the devs, but I haven't looked tbf. Not sure where the 20 games limit stems from, I don't really understand.

**Reply by malice00** [reply]
20 because you can only use letters A-Z (so 26) and like I said there are already 4 drives mounted and (possibly) A and B can't be used -- so 20.
OK, well, I hope official word comes at some point...

**Reply by hahayupgit** [reply]
Just posted a response that should answer these questions! Let me know if I can provide any clarification. You can absolutely have more than 20 games, you definitely don't need to add an individual drive for each game.

**Comment by hahayupgit** [comment, marked as answer]
TL;DR you are recommended to create new bottles for each game, install Steam within, and connect to a unified "drive" that hosts all your games. This way, you can install specific fixes, winetricks, and launch options for each game in a way that doesn't conflict with other games. That being said, having one bottle with Steam also generally works perfectly fine.

The method posted by billiamg works, however you don't _need_ to do any configuration with Drives in the Wine configuration. 

The method that I use is to create a folder somewhere that you'd want to store all your Steam games. My folder is located in `~/Library/Steam Library`.

After this, I install Steam normally in a Whisky bottle (as in installing Steam without any special tweaks to the bottle), and then open Steam, go to `Steam -> Settings -> Storage`, click on `Local Drive` and click the `Add Drive` option.

In the popup, navigate to the folder you created above, and click `Open`. This adds that folder as a "drive" within Steam, and enables you to install games to this location. 

The reason this is particularly useful is that this drive can be used to store games in a place where multiple different bottles can access. This also enables you to create a unified drive that can be accessed from CrossOver, Wineskin (now Kegworks) and more. 

Additionally, this means that if something goes wrong with your bottle (corruption, Steam breaking, etc) then you can easily delete that bottle and create a new one without having to re-install any of your games.

**Reply by johnboiles** [reply]
Do you use the Z: drive to access your macOS drive to mount your `~/Library/Steam Library` folder or is there a more correct way?

**Reply by hahayupgit** [reply]
Yes, you use the Z: drive! I'm not aware of a more correct way, but even if there was a more "correct" way it would still get you the same result.

**Reply by scaryrawr** [reply]
Do you install steam in each bottle (multiple steam installs)? And they all share the same shared drive for the Steam library?
